conf4j-core/src/main/java/com/sabre/oss/conf4j/converter/AbstractTemporalAccessorConverter.java

```diff
@@ -0,0 +1,138 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.conf4j.converter;
+
+import com.sabre.oss.conf4j.internal.utils.spring.ConcurrentReferenceHashMap;
+
+import java.lang.reflect.Type;
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This is base class for converting date-time values from/to string.
+ * <p>
+ * The converter supports {@value #FORMAT} meta-attribute which specifies the format of resulting string representation.
+ * The format string must be compatible with the format defined by {@link DateTimeFormatter}.
+ * <p>
+ * While converting from value to string if the format is not provided,
+ * appropriate ISO representation is used (for example {@link DateTimeFormatter#ISO_LOCAL_DATE_TIME})
+ *
+ * @param <T> actual temporal accessor type
+ */
+
+public abstract class AbstractTemporalAccessorConverter<T extends TemporalAccessor> implements TypeConverter<T> {
+    /**
+     * Format attribute name.
+     */
+    public static final String FORMAT = "format";
+
+    private static final ConcurrentMap<String, DateTimeFormatter> cache = new ConcurrentReferenceHashMap<>();
+
+    protected abstract T parse(String value, DateTimeFormatter formatterForPattern);
+
+    protected abstract DateTimeFormatter getDefaultFormatter();
+
+    /**
+     * Converts String to {@code T}.
+     *
+     * @param type       actual type definition.
+     * @param value      string representation of the value which is converted to a given type.
+     *                   In case it is {@code null}, the converter should return {@code null}.
+     * @param attributes additional meta-data attributes which may be used by converter. It can be {@code null}.
+     *                   If present, the value for {@value #FORMAT} key will be used during conversion
+     *                   as a formatting pattern.
+     * @return value converted to {@code T}
+     * @throws IllegalArgumentException when {@code value} cannot be converted to {{@code T} because of
```

Comment:
Here is small issue in javadoc here - there is double angle bracket in the code `{{@code T}` instead of one.